website/docs/pubsub.md

```diff
@@ -43,185 +43,191 @@ data:
-        dapr.io/app-port: "6002"
     spec:
-      containers:
-      - name: go-sub
-        image: fake-subscriber:latest
-        imagePullPolicy: Never
-```
-**Note:** Dockerfile to build image for fake-subscriber is under [gatekeeper/test/fake-subscriber](https://github.com/open-policy-agent/gatekeeper/tree/master/test/pubsub/fake-subscriber). You can find make rule to build and deploy subscriber in [Makefile](https://github.com/open-policy-agent/gatekeeper/blob/master/Makefile) under name `e2e-subscriber-build-load-image` and `e2e-subscriber-deploy`.
+      replicas: 1
+      selector:
+        matchLabels:
+          app: sub
+      template:
+        metadata:
+          labels:
+            app: sub
+          annotations:
+            dapr.io/enabled: "true"
+            dapr.io/app-id: "subscriber"
+            dapr.io/enable-api-logging: "true"
+            dapr.io/app-port: "6002"
+        spec:
+          containers:
+          - name: go-sub
+            image: fake-subscriber:latest
+            imagePullPolicy: Never
+    ```
+
+    **Note:** Dockerfile to build image for fake-subscriber is under [gatekeeper/test/fake-subscriber](https://github.com/open-policy-agent/gatekeeper/tree/master/test/pubsub/fake-subscriber).
 
 #### Configure Gatekeeper with Pubsub enabled
 
 1. Create Dapr pubsub component and Redis secret in Gatekeeper's namespace (`gatekeeper-system` by default). Please make sure to update `gatekeeper-system` namespace for the next steps if your cluster's Gatekeeper namespace is different.
 
-```shell
-kubectl get secret redis --namespace=default -o yaml | sed 's/namespace: .*/namespace: gatekeeper-system/' | kubectl apply -f -
-kubectl apply -f - <<EOF
-apiVersion: dapr.io/v1alpha1
-kind: Component
-metadata:
-  name: pubsub
-  namespace: gatekeeper-system
-spec:
-  type: pubsub.redis
-  version: v1
-  metadata:
-  - name: redisHost
-    value: redis-master.default.svc.cluster.local:6379
-  - name: redisPassword
-    secretKeyRef:
-      name: redis
-      key: redis-password
-EOF
-```
-
-2. Install Gatekeeper with `--enable-pub-sub` set to `true`, `--audit-connection` set to `audit-pubsub-connection`, `--audit-channel` set to `audit` on audit pod.
-
-
-```shell
-echo 'auditPodAnnotations: {dapr.io/enabled: "true", dapr.io/app-id: "audit", dapr.io/metrics-port: "9999"}' > .tmp/annotations.yaml # auditPodAnnotations is used to add annotations required by Dapr to inject sidecar to audit pod
-helm install gatekeeper/gatekeeper --name-template=gatekeeper --namespace gatekeeper-system \
---set audit.enablePubsub=true \
---set audit.connection=audit-pubsub-connection \
---set audit.channel=audit \
---values .tmp/annotations.yaml
-```
-
-**Note:** Verify that after the audit pod is running there is a dapr sidecar injected and running along side `manager` container.
+    ```shell
+    kubectl get secret redis --namespace=default -o yaml | sed 's/namespace: .*/namespace: gatekeeper-system/' | kubectl apply -f -
```

Comment:
if we haven't installed gatekeeper yet (it's in next step), `gatekeeper-system` namespace will not exist.

can we swap these steps?

Reply:
`Component` needs to exist in the namespace before gk pods and sidecar are initialized. I will update first step to `Create gk namespace and dapr component`. Does that work?